docs: Add copy code button to portico docs.

josiah-tesfu opened the pull request:
Added a copy-to-clipboard button to the code blocks in the API and Help Center docs. Previously, copying code from the docs required manual copying, which was cumbersome.

Used the same copy-to-clipboard svg icon as the one used in web/src but manually created the button within the js function instead of using a template. Updated the `pre` CSS element to have relative positioning and gave the `copy-codeblock` element absolute positioning to ensure the button stayed in the top-right corner.

Fixes: #25726.

**Screenshots and screen captures:**
![issue-25762-record](https://github.com/zulip/zulip/assets/71205057/0159099e-cdc6-436c-bb98-bc3b6a62c7ba)




<details>
<summary>Self-review checklist</summary>

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

josiah-tesfu commented:
Ready for review.

palashb01 commented:
@josiah-tesfu , it's always better to include the entire screenshot. Currently, the tooltip of the button is not visible. Adding the complete screenshot, including the tooltip, provides comprehensive information, making it easier for the reviewer to review as well.

josiah-tesfu commented:
Updated the commit message and changed the screen recording.

palashb01 commented:
@josiah-tesfu , the screenshot looks great. However, in Zulip, we highly recommend wrapping lines after 69 characters (max 71). Therefore, I would suggest rebasing the commit description and wrapping all the lines so that none of them exceeds 69 characters

josiah-tesfu commented:
Fixed the line-wrap on the commit message.

karlstolley commented on web/styles/portico/markdown.css:
```diff
@@ -392,6 +396,19 @@
+        background: none;
+        border: none;
+        padding: 0;
+        display: inline-block;
```
It should be safe to strike the `inline-block` line here. Buttons display as `inline-block` by default, but with the absolute positioning here, there should be no effect from this property.

karlstolley commented:
One thing I noticed in testing is that there is a likely browser-default outline around the button when you reach it via the `Tab` key:

<img width="736" alt="Screenshot 2023-05-29 at 2 27 58 PM" src="https://github.com/zulip/zulip/assets/170719/365bec91-756a-4387-8064-7fa42855d279">

It's good accessibility practice to provide a `:focus` indicator like that, but you might look at adjusting the padding some and using one of the colors in use on the page, like `hsl(176deg 46% 41%)`, to give this a bit more finish--especially because you can activate the button from the keyboard.

josiah-tesfu commented:
@karlstolley, thanks for the review. I updated the code with the changes you suggested.

josiah-tesfu commented:
Ready for review.

timabbott commented on web/src/portico/help.js:
```diff
@@ -56,6 +91,11 @@ function render_code_sections() {
 
+    // Add a copy-to-clipboard button for each pre element
+    $(".markdown pre").each(function () {
+        copyCodeToClipboard($(this));
```
We should probably call this `add_copy_to_clipboard_element`

timabbott commented on web/src/portico/help.js:
```diff
@@ -27,6 +30,38 @@ function registerCodeSection($codeSection) {
+        // Show the tooltip for 1s
+        setTimeout(() => {
+            tooltip.hide();
+        }, 1000);
```
Doesn't the `duration` parameter take care of this?

timabbott commented on web/styles/portico/markdown.css:
```diff
@@ -382,6 +382,10 @@
     & pre {
+        /* Relative positioning ensures the copy-to-
+        clipboard button remains in the top-left corner */
+        position: relative;
```
Does this have any side effects?

josiah-tesfu replied:
No, I don't think so. From what I learned, relative positioning only changes an element if positioning constraints are specified.

shameondev commented:
Hey, @josiah-tesfu. It's a good practice to give a brief summary of the changes you made in the last review iteration before requesting a review. This way, it's easier for the reviewer to help us moving forward! You can read more about it in the docs: [How to help move the review process forward](https://zulip.readthedocs.io/en/latest/contributing/review-process.html#how-to-help-move-the-review-process-forward).

Would you be up for doing this?

josiah-tesfu commented:
Review update:
- Changed function name from `copyCodeToClipboard` to `add_copy_to_clipboard_element`
- Removed the `duration` parameter from the tippy tooltip

timabbott commented on web/styles/portico/markdown.css:
```diff
@@ -392,6 +396,24 @@
+        border: none;
+        padding: 0;
+        position: absolute;
+        /* Position the button in the top-left corner */
```
These comments twice mention top-left, when it's the top-right corner; I just fixed this before merging.

timabbott commented:
Merged, as this checks off the feature, but opened https://github.com/zulip/zulip/issues/25962 for some hopefully pretty straightforward polish that I think needs as a follow-up.
